File: MeetSpaceBackend/MeetSpace.Services/Database/Entities/Facility.cs
namespace MeetSpace.Models.Entities
{
    public class Facility
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public int CityId { get; set; }
        public string? Description { get; set; }
        public string? ContactEmail { get; set; }
        public string? ContactPhone { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation
        public virtual City City { get; set; }
        public virtual ICollection<Space> Spaces { get; set; }

        public Facility()
        {
            Spaces = new HashSet<Space>();
        }
    }
}

File: MeetSpaceFrontend/meetspace_desktop/lib/models/facility.dart
class Facility {
  final int id;

  final String name;
  final String address;

  final int cityId;

  final String cityName;
  final String countryName;

  final String? description;
  final String? contactEmail;
  final String? contactPhone;

  Facility({
    required this.id,
    required this.name,
    required this.address,
    required this.cityId,
    required this.cityName,
    required this.countryName,
    this.description,
    this.contactEmail,
    this.contactPhone,
  });

  factory Facility.fromJson(
    Map<String, dynamic> json,
  ) {
    return Facility(
      id: json["id"],

      name: json["name"] ?? "",

      address: json["address"] ?? "",

      cityId: json["cityId"],

      cityName:
          json["cityName"] ?? "",

      countryName:
          json["countryName"] ?? "",

      description:
          json["description"],

      contactEmail:
          json["contactEmail"],

      contactPhone:
          json["contactPhone"],
    );
  }
}